thirdeye/thirdeye-pinot/src/main/java/org/apache/pinot/thirdeye/detection/spi/model/TimeSeries.java

```diff
@@ -53,17 +53,47 @@ public void addPredictedBaseline(DoubleSeries baselineValues) {
     this.df.addSeries(DataFrameUtils.COL_VALUE, baselineValues);
   }
 
+  /**
+   * Add the predicted upper bound into the timeseries
+   * @param upperBoundValues predicted upper bound values
+   */
+  public void addPredictedUpperBound(DoubleSeries upperBoundValues) {
```

Comment:
I'd prefer we have a time series builder class to construct the TimeSeries class. Something like:

` Public static class Builder() { `
        `  public Builder(LongSeries timeStamp, DoubleSeries currentTimeSeries, DoubleSeries predictedBaseline); `
       `   public Builder upperBound(DoubleSeries predictedUpperBound);`
       `   public Builder lowerBound(DoubleSeries predictedLowerBound);`
       ` .  public TimeSeries Build();`
`   }
`
Then the DataFrame class won't be exposed to client and we have a explicit contract to build TimeSeries instance.

Reply:
As discussed. Changed the constructors to handle it more efficiently.